Articles/Blog/HelloHeatEquation.md

```diff
@@ -0,0 +1,226 @@
+H1_FECollection fe_coll(order, dim);
+FiniteElementSpace fespace(mesh, &fe_coll);
+FunctionCoefficient u_0(InitialTemperature);
+u_gf.ProjectCoefficient(u_0);
+Vector u;
+u_gf.GetTrueDofs(u);
+ConductionOperator oper(fespace, alpha, kappa, u);
+
+// 5. Perform time-integration
+ode_solver->Init(oper);
+double t = 0.0;
+bool last_step = false;
+for (int ti = 1; !last_step; ti++){
+// Time integration at each step
+}
+
+// 6. Save the final solution that can be viewed later
+ofstream osol("ex16-final.gf");
+osol.precision(precision);
+u_gf.Save(osol);
+
+```
+
+This implementation clearly demonstrates the complexity introduced by using a third-party library. However, users gain access to a number of numerical solvers, parallelism, and even GPU capability.
+
+### Library Example 2: FD1D\_HEAT\_EXPLICIT (C)
+While highly developed numerical libraries such as MFEM have a wide range of functionality and support options, others are smaller. While they still provide support for different use cases of function like the heat equation, they do not come which the obscurity and extra features that may intimidate users.
+
+[FD1D\_HEAT\_EXPLICIT](https://people.sc.fsu.edu/~jburkardt/c_src/fd1d_heat_explicit/fd1d_heat_explicit.html "FD1D\_HEAT\_EXPLICIT") is a simple 1-dimensional Heat Equation solver which uses the explicit version of the finite difference method to handle time integration. Versions of this library are written in C, C++, Fortran90, Matlab, and Python. The code example from the library shown below is sequential, but can easily be adapted to become parallelized.
+
+```c
+double *fd1d_heat_explicit ( int x_num, double x[], double t, double dt, 
+  double cfl, double *rhs ( int x_num, double x[], double t ), 
+  void bc ( int x_num, double x[], double t, double h[] ), double h[] )
+  {
+  double *f;
+  double *h_new;
+  int j;
+
+  f = rhs ( x_num, x, t );
+
+  h_new = ( double * ) malloc ( x_num * sizeof ( double ) );
+
+  h_new[0] = 0.0;
+
+  for ( j = 1; j < x_num - 1; j++ )
+  {
+    h_new[j] = h[j] + dt * f[j] 
+      + cfl * (         h[j-1]
+                - 2.0 * h[j]
+                +       h[j+1] );
+  }
+  h_new[x_num-1] = 0.0;
+
+  bc ( x_num, x, t + dt, h_new );
+
+  free ( f );
+
+  return h_new;
+}
+```
+
+While this code is almost identical the hand-coded example shown above, this library does include some extra features that allows users to implement some additional functionality including writing results to different kinds of R8 files, adding a timestamp, and solving for the Courant-Friedrichs-Loewy coefficient.
+
+### More Implementations!
+Using the Heat Equation to demonstrate the power of a programming model or mathematical library is not new. In fact, it is common tutorial exercise for new scientific programmers. Since there are too many interesting examples to include in this article, we've started [a repository showcase](https://github.com/betterscientificsoftware/hello-heat-equation) to collect the many examples.
```

Comment:
This is great. Maybe some instructions to people on how to submit a new implementation. We should think about this a bit too. We might want all implementations to share some common base code and behavior.

Reply:
Having a contributing guide is a great idea! I want to stay away from having a common base since this code is not runnable (and making it so would be way too complex). We should point to your CI-examples for runnable code.

Reply:
We are working on a contributing guide for that repo. I'll mark this as resolved here.